### PR TITLE
Pn 45 user card

### PR DESCRIPTION
**Pull request by joezheng416**
image
Finish the pet sitter list component.
Next step: add responsive and pagination

### CODE REVIEW

File: src/components/userCard/PetSitterCard/PetSitterCard.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  CardFooterText,
+} from "./StyledPetSitterCard";
+
+const PetSitterCard = (prop: PetSitter) => {
+  return (
+    <PetSitterCardContainer>
+      <CardHeader>
+        <PetSitterCardHeaderContainer>
+          <CardHeaderLeft>
+            <UserAvatar name={prop.name} src={prop.avatar} />
```
Comment (jube33717f) [inline src/components/userCard/PetSitterCard/PetSitterCard.tsx:23]: destructuring `const PetSitterCard = ({name,avatar,...}: PetSitter)`
ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

File: src/components/userCard/PetSitterCard/PetSitterCard.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          </CardHeaderLeft>
+          <Box>
+            <PriceInfo>
+              {prop.price} AUD <br></br>/ night{" "}
+            </PriceInfo>
+          </Box>
+        </PetSitterCardHeaderContainer>
+      </CardHeader>
+      <CardBody>
+        <CardIntroduction>{prop.introduction}</CardIntroduction>
```
Comment (jube33717f) [inline src/components/userCard/PetSitterCard/PetSitterCard.tsx:48]: same before

File: src/components/userCard/PetSitterCard/PetSitterCard.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    .map((_, i) => (
+                      <StarIcon key={i} color={i < prop.rating ? "yellow.500" : "gray.300"} /> //show stars according to petSitter rates
+                    ))}
+                </div>
+              </CardHeaderMiddle>
+            </Box>
+          </CardHeaderLeft>
+          <Box>
+            <PriceInfo>
+              {prop.price} AUD <br></br>/ night{" "}
```
Comment (jube33717f) [inline src/components/userCard/PetSitterCard/PetSitterCard.tsx:42]: better to use style to implement

File: src/components/userCard/PetSitterCardList/PetSitterCardList.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          id={petSitter.id}
+          name={petSitter.name}
+          avatar={petSitter.avatar}
+          suburb={petSitter.suburb}
+          price={petSitter.price}
+          introduction={petSitter.introduction}
+          distance={petSitter.distance}
+          rating={petSitter.rating}
+        />
+      ))}
```
Comment (jube33717f) [inline src/components/userCard/PetSitterCardList/PetSitterCardList.tsx:92]: destructuring

**Review by jube33717f — CHANGES_REQUESTED** [review]

**Review by ShawnWang86b — APPROVED** [review]

**Review by applestory — APPROVED** [review]
export const CardHeaderLeft = styled(Flex)`
   &&& {
     gap: 4;
   }
 `;

I am not sure whether the gap:4 should have a unit or not, 4rem or 4px.

**Comment by joezheng416** [comment]
> export const CardHeaderLeft = styled(Flex)`&&& { gap: 4; }`;
> 
> I am not sure whether the gap:4 should have a unit or not, 4rem or 4px.

Thanks Dylan. This should be 16px. Updated & committed

**Comment by joezheng416** [comment]
> export const CardHeaderLeft = styled(Flex)`&&& { gap: 4; }`;
> 
> I am not sure whether the gap:4 should have a unit or not, 4rem or 4px.

Thanks Dylan. This should be 16px. Updated & committed via 6bbc5c5

**Pull request merged**